src/sage/graphs/graph.py

```diff
@@ -8209,8 +8209,7 @@ def modular_decomposition(self, algorithm=None, style='tuple'):
         if not self.order():
             D = None
         elif self.order() == 1:
-            D = create_prime_node()
-            D.children.append(create_normal_node(self.vertices(sort=False)[0]))
+            D = create_normal_node(self.vertices(sort=False)[0])
```

Comment:
you could use `D = create_normal_node(next(self.vertex_iterator()))`

Reply:
done in 420692b